src/cpp/benders/benders_core/BendersBase.cpp

```diff
@@ -931,16 +963,87 @@ WorkerMasterData BendersBase::BestIterationWorkerMaster() const {
   return relevantIterationData_.best;
 }
 
-void BendersBase::ResetData(double criterion) {
+void BendersBase::InitExternalValues(bool is_bilevel_check_all, double lambda) {
+  // _data.outer_loop_current_iteration_data.outer_loop_criterion = 0;
+  // _data.outer_loop_current_iteration_data.benders_num_run = 1;
+  is_bilevel_check_all_ = is_bilevel_check_all;
+  outer_loop_biLevel_.Init(MasterObjectiveFunctionCoeffs(),
+                           BestIterationWorkerMaster().get_max_invest(),
+                           MasterVariables());
+  outer_loop_biLevel_.SetLambda(lambda);
+}
+
+CurrentIterationData BendersBase::GetCurrentIterationData() const {
+  return _data;
+}
+OuterLoopCurrentIterationData BendersBase::GetOuterLoopData() const {
+  return _data.outer_loop_current_iteration_data;
+}
+std::vector<double> BendersBase::GetOuterLoopCriterionAtBestBenders() const {
+  return ((outer_loop_criterion_.empty())
+              ? std::vector<double>()
+              : outer_loop_criterion_[_data.best_it - 1]);
+}
+
+std::vector<double> BendersBase::ComputeOuterLoopCriterion(
+    const std::string &subproblem_name,
+    const PlainData::SubProblemData &sub_problem_data) {
+
+  auto outer_loop_input_size = var_indices_.size(); // num of patterns
+  std::vector<double> outer_loop_criterion_per_sub_problem(outer_loop_input_size,
+                                                           {});
+  auto subproblem_weight = SubproblemWeight(_data.nsubproblem, subproblem_name);
+  double criterion_count_threshold =
```

Comment:
To be removed, unnecessary

Reply:
why? it's used to compute outer_loop_criterion_per_sub_problem

Reply:
My comment was not intended to be on these lines ^^